### PR TITLE
Upgrade dependencies and introduce unexpectedjs

### LINKED ISSUE #171: Old package warnings on install
Upon install:

```
$ npm i -g create-elm-app
npm WARN deprecated babel-preset-es2017@6.24.1: We're really excited that you're trying to use ES2017 syntax, but instead of making more yearly presets, Babel has a better preset that we recommend you use instead: npm install babel-preset-env --save-dev. Give us a follow @babeljs to get more updates
npm WARN deprecated babel-preset-es2016@6.24.1: We're really excited that you're trying to use ES2016 syntax, but instead of making more yearly presets, Babel has a better preset that we recommend you use instead: npm install babel-preset-env --save-dev. Give us a follow @babeljs to get more updates
npm WARN deprecated node-uuid@1.4.8: Use uuid module instead
```

It would be great to see these go away - probably create-elm-app depends on an old version of something that depends on an old version of these packages.

### PR DESCRIPTION
**Pull request by halfzebra**
Closes #171

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -44,29 +44,29 @@
...
     "husky": "^0.14.3",
-    "mocha": "^3.5.0",
+    "mocha": "^3.5.2",
     "nightmare": "^2.10.0",
-    "prettier": "^1.5.3",
+    "prettier": "^1.6.1",
     "rimraf": "^2.5.4",
-    "semantic-release": "^6.3.2"
+    "semantic-release": "^7.0.2",
+    "unexpected": "^10.35.0"
```
Comment (fobos) [inline package.json]: For what reason you have changed `chai` to `unexpected`?

File: package.json [hunk 2]
```diff
@@ -93,8 +93,8 @@
     "eslint": "eslint .",
     "prettier": "prettier --write --single-quote '{bin,config,scripts,template/src,tests}/**/*.js'",
     "semantic-release": "semantic-release pre && npm publish && semantic-release post",
-    "test": "npm run -s test:cli && npm run -s test:functional",
-    "test:cli": "mocha tests/cliAccessibility.js --reporter spec --timeout 15000",
-    "test:functional": "mocha tests/*.spec.js --reporter spec --timeout 15000"
+    "test": "npm run test:cli && npm run test:functional",
```
Comment (fobos) [inline package.json]: Why do we need odd output?

Comment (halfzebra) [reply package.json]: That's clearly a mistake!

File: package.json (hunk 1 above)
Comment (halfzebra) [reply package.json]: @fobos it produces more readable test errors, obviously, we don't have that many tests, so it's a little smaller as well.

I would roll back if you have a strong opinion on assertion lib. I've been doing a lot of testing with unexpected recently and I would say I've had a better time :slightly_smiling_face:

Comment (fobos) [reply package.json]: Thanks for explanation. 

> I would roll back if you have a strong opinion on assertion lib.
It's ok to use `unexpected`. I see your points.

**Pull request merged**